backend/pium/src/main/java/com/official/pium/controller/GlobalExceptionHandler.java

```diff
@@ -34,26 +37,50 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotV
             stringJoiner.add(error.getDefaultMessage());
         }
 
-        GlobalExceptionResponse globalExceptionResponse = createExceptionResponse(stringJoiner.toString());
+        String message = stringJoiner.toString();
+        GlobalExceptionResponse globalExceptionResponse = createExceptionResponse(message);
+        log.info(message);
         return ResponseEntity.badRequest().body(globalExceptionResponse);
     }
 
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<GlobalExceptionResponse> handleException(Exception e) {
+        String message = e.getMessage();
+        GlobalExceptionResponse exceptionResponse = createExceptionResponse(SERVER_ERROR_MESSAGE);
+        log.error(message);
+        return ResponseEntity.internalServerError().body(exceptionResponse);
```

Comment:
### C

에러를 로깅하는 이유는 어디에서 어떤 문제가 발생했는지 빠르게 파악하기 위함이라고 생각해요!

그런 의미에서 getMessage()는 로그 자료로써  그닥 좋은 내용은 아니라는 생각이 듭니다.

Exception은 stackTrace를 로깅하는것이 어떨까요? (슬렉 메시지도 StackTrace가 오도록)

```text
// getMessage로 기록되는 정보

2023-08-13 17:23:52.627 [http-nio-8080-exec-1] ERROR com.official.pium.controller.GlobalExceptionHandler - Cannot invoke "String.length()" because "a" is null
```


```text
// stackTrace로 기록되는 정보

2023-08-13 17:42:49.067 [http-nio-8080-exec-1] ERROR com.official.pium.controller.GlobalExceptionHandler - 서버에 문제가 발생했어요!!
java.lang.NullPointerException: Cannot invoke "String.length()" because "a" is null
	at com.official.pium.controller.HistoryController.test(HistoryController.java:43)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:141)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.official.pium.controller.HistoryController$$SpringCGLIB$$0.test(<generated>)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
...
...
```

1. String 변환 하는 방법
```suggestion
        StringWriter stringWriter = new StringWriter();
        e.printStackTrace(new PrintWriter(stringWriter));
        String message = stringWriter.toString();
        log.error(message);

        GlobalExceptionResponse exceptionResponse = createExceptionResponse(SERVER_ERROR_MESSAGE);
        return ResponseEntity.internalServerError().body(exceptionResponse);
```

2. log.error의 인자 활용
```suggestion
        
        log.error("서버에 문제가 발생했어요!!", e);

        GlobalExceptionResponse exceptionResponse = createExceptionResponse(SERVER_ERROR_MESSAGE);
        return ResponseEntity.internalServerError().body(exceptionResponse);
```

Reply:
좋습니다 !
신경쓰지 않고 있었던 부분인데 잘 짚어주셨네요 ㅎㅎ

`ERROR` 레벨은 말씀해주신 대로 적용하면 될 것 같아요. 그리고 방법으로는 2번 방법이 더 좋을 것 같습니다 !